src/main.rs

```diff
@@ -125,242 +125,307 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
-        penalty_add_upper_layout.attach(&delete_time_penalty, 0, 4, 1, 1);
-        penalty_add_upper_layout.attach(&penalty_player_number, 1, 4, 1, 1);
-        penalty_add_upper_layout.attach(&start_time_penalty, 2, 4, 1, 1);
-        penalty_add_upper_layout.attach(&penalty_1min, 2, 0, 1, 1);
-        penalty_add_upper_layout.attach(&penalty_2min, 2, 1, 1, 1);
-        penalty_add_upper_layout.attach(&penalty_5min, 2, 2, 1, 1);
-        penalty_add_upper_layout.attach(&penalty_dismiss, 2, 3, 1, 1);
-
-        let timeout_ribbon = new_timeout_ribbon();
-
-        penalty_add_layout.pack_start(&penalty_add_upper_layout, false, false, 0);
-        penalty_add_layout.pack_start(&timeout_ribbon, false, false, 0);
+        penalty_add_layout.attach(&penalty_keypad, 0, 0, 4, 7);
+        penalty_add_layout.attach(&penalty_white_select, 4, 0, 4, 3);
+        penalty_add_layout.attach(&penalty_black_select, 8, 0, 4, 3);
+        penalty_add_layout.attach(&penalty_1min, 4, 3, 2, 4);
+        penalty_add_layout.attach(&penalty_2min, 6, 3, 2, 4);
+        penalty_add_layout.attach(&penalty_5min, 8, 3, 2, 4);
+        penalty_add_layout.attach(&penalty_dismiss, 10, 3, 2, 4);
+        penalty_add_layout.attach(&penalty_delete, 0, 7, 4, 2);
+        penalty_add_layout.attach(&penalty_player_number, 4, 7, 4, 2);
+        penalty_add_layout.attach(&penalty_add, 8, 7, 4, 2);
+        penalty_add_layout.attach(&penalty_white_timeout, 0, 9, 3, 2);
+        penalty_add_layout.attach(&penalty_referee_timeout, 3, 9, 6, 2);
+        penalty_add_layout.attach(&penalty_black_timeout, 9, 9, 3, 2);
 
         // Time Edit Page
         let time_edit_layout = gtk::Grid::new();
-        time_edit_layout.insert_column(0);
-        time_edit_layout.insert_column(1);
-        time_edit_layout.insert_column(2);
-        time_edit_layout.insert_row(0);
-        time_edit_layout.insert_row(1);
-        time_edit_layout.insert_row(2);
+        time_edit_layout.set_column_homogeneous(true);
+        time_edit_layout.set_row_homogeneous(true);
+        time_edit_layout.set_margin_top(BUTTON_MARGIN);
+        time_edit_layout.set_margin_start(BUTTON_MARGIN);
+        time_edit_layout.set_margin_end(BUTTON_MARGIN);
+        time_edit_layout.set_margin_bottom(BUTTON_MARGIN);
         time_edit_layout.set_column_spacing(BUTTON_SPACING.try_into().unwrap());
         time_edit_layout.set_row_spacing(BUTTON_SPACING.try_into().unwrap());
 
-        let _minute_plus = new_button("+", "gray", None);
-        let _minute_minus = new_button("-", "gray", None);
-        let _second_plus = new_button("+", "gray", None);
-        let _second_minus = new_button("-", "gray", None);
+        let minute_plus = new_button("+", "blue", None);
+        let minute_minus = new_button("-", "light-blue", None);
+        let second_plus = new_button("+", "blue", None);
+        let second_minus = new_button("-", "light-blue", None);
```

Comment:
why are these different styles? esp different text styles

Reply:
I didn't pay much attention to text size yet, I was hoping there could be some way to auto-scale them to the window size. 

They are different styles just to differentiate them a little bit, they probably could be the same but I thought it might be nice to have them different colors

Reply:
not sure I like the different colors, but doesn't really bother me I guess. As for text size, I don't know of an easy way to scale that.

Reply:
This can be fixed in a follow up PR